Fix trading form input field issues

ebma opened the pull request:
Fixes issues on iOS where the user was unable to enter a valid amount because the on-screen keyboard only showed a comma which was not accepted by the validation mechanism (when the device is set up for EU regions).

Seems like this issue appears on iOS only. When entering a value with a comma into the input field (either the primary amount or manual price, does not matter) it is not propagated correctly, i.e. the onChange event only contains `""`. Removing the `type: "number"` prop helps here, but this is not enough as `Big.js` somehow still has issues with the comma string and throws an `Invalid number` error when trying to instantiate a new `Big()`. That's where the `replaceCommaWithDot(input:string)` function comes into play. 

Also closes #1251 by re-introducing the validation rule such that the price input field will already show an "Invalid amount" error instead of first showing it when the user clicks on the submit button.

andywer commented on src/Trading/components/TradingPrice.tsx:
```diff
@@ -64,7 +64,6 @@ const TradingPrice = React.forwardRef(function TradingPrice(props: TradingPriceP
       onChange={props.onChange}
       onFocus={props.selectOnFocus ? event => event.target.select() : undefined}
       style={props.style}
-      type="number"
```
@ebma Do we still get to see the numpad rather than the default keyboard? 😉

ebma replied:
Yes, at least on iOS (didn't check on Android). I guess it's related to the pattern `pattern: "[0-9]*",` and not necessarily the `type="number"`.

andywer commented on src/Trading/hooks/form.ts:
```diff
@@ -25,6 +25,12 @@ function getSpendableBalanceWithoutBaseReserve(accountMinimumBalance: BigNumber,
+// replaces ',' with '.' in a string
+// this can be used with strings that represent a number before passing them to Big()
+export function replaceCommaWithDot(input: string) {
+  return input.replace(/,/g, ".")
```
Is there anything we can do to prevent accidents in the case that someone tries to enter 3000 as `3,000`?

Bonus question: Does the form do its job if I enter 0.25 as `.25`?

andywer replied:
One more thing:
Now is logic is custom for the trading view. Actually, it should be consistent across all our forms… 🙈

ebma replied:
Regarding the bonus question: yes it would work if we change the regex pattern. Right now it will expect something like `x.xx` but seems like Big.js can also use strings without leading 0.
And I don't think we can do much about the `3,000` meaning 3000 or at least not without much effort. Seems like this locale thing is quite complex and to be honest I still don't fully understand why iOS is causing this issue anyways.

ebma replied:
I added the additional handling to the payment form as well, and I think the trading and payment form would be the only places where we actually expect inputs with fractions? So it should be fine, but I see your point that we could spend more efforts to fully prevent this issue in the future.

ebma commented:
I moved the new utility functions to a new file and created a new function called `FormBigNumber()` which should be used instead of `BigNumber()` whenever we are instantiating a Big with user input. 
And I also changed the regex such that `.xx` or `,xx` are also accepted.
